### DISCUSSION TITLE
Apparent Performance Issues with Logstash on the Manager

### DISCUSSION THREAD

**Discussion by jimhranicky** [Security-Onion-Solutions/securityonion#6388] [Unsupported Versions]
Current version: (currently 2.3.90 as of today, issue discovered on 2.3.80)
Deployment: on-prem
Installation method: ISO
Nodes: 1 manager (2cpu/36c/72t), 1 sensor (1cpu/64c/128t), 9 storage (VMs)
Monitoring: Tap, 10G
so-status:
    manager : All services running
    sensor  : All services running except strelka (probably due to our attempt to disable)
    storage : All services running
No salt errors
Grid: Only shows errors related to strelka on the sensor
Issue:

With the default installation, filebeat cannot seem to keep up with the number of events generated by zeek/suri on our sensor. Changes to filebeat/logstash configs on the sensor/manager/storage nodes were unsuccessful.

I wrote a simple script to read the filebeat logs to see how much of each file was being read. conn and dns logs were regularly far behind, filebeat could only read 2-10% of the files before rotation. filebeat would then continue to read the rotated file, however, one or more hours would pass during this time, and other files would get rotated before ever having been read, leading to a "spiky triangle" graph in kibana.

We decided to remove the manager ingest logstash from the equation and have filebeat just send straight to redis. This led to a huge jump in performance, and it seems filebeat can now keep up with the events generated on the sensor. The EPS on our sensor went from 14-15K to 28-32K after the change.

During the course of our troubleshooting, it was difficult to determine where the bottleneck was - was filebeat itself unable to keep up? Was there backpressure in the pipeline (this appears to have been the case)? What was the cause of the backpressure? We tried turning various knobs until we stumbled on a solution that seems to be working at the moment. Is there a better way to go about troubleshooting such issues?

Also, should we revisit the settings for the manager logstash, or just continue sending FB events straight to redis?

Thanks.

**Comment by weslambert** [comment]
You might try adjusting the Logstash thread/worker values, in addition to the Redis output batch count settings to try to find the sweet spot for pushing to and pulling from Redis.

Ex. `ls_pipeline_batch_size` in the manager's sls file.
